tensorflow/python/distribute/cluster_resolver/slurm_cluster_resolver.py

```diff
@@ -95,20 +126,33 @@ def __init__(self,
       num_tasks = int(os.environ['OMPI_COMM_WORLD_SIZE'])
     else:
       self._rank = int(os.environ['SLURM_PROCID'])
-      num_tasks = int(os.environ['SLURM_NTASKS'])
+      num_tasks = get_num_slurm_tasks()
+
+    if jobs is None:
+      jobs = {'worker': num_tasks}
+    if gpus_per_node is None:
+      gpus_per_node = sum(l.startswith('GPU ') for l in
+                          subprocess.check_output(['nvidia-smi', '--list-gpus']).
```

Comment:
Can we abstract this out to another function? The function should also throw an exception if `nvidia-smi` is not found for instance.

Reply:
Sure